lib/chef/resource/windows_certificate.rb

```diff
@@ -309,11 +309,7 @@ def acl_script(hash)
         # @raise [OpenSSL::PKCS12::PKCS12Error] When incorrect password is provided for PFX certificate
         #
         def fetch_cert_object(ext)
-          contents = if binary_cert?
-                       ::File.binread(new_resource.source)
-                     else
-                       ::File.read(new_resource.source)
-                     end
+          contents = ::File.binread(new_resource.source)
```

Comment:
Reading non-binary certs with `File.binread` appears to cause no effect, so avoid shelling out to find out if the file is binary. This is already done in other places (#10751), so it seems safe to do here as well.